File: Conway's Game of Life/[VB]-Conway's Game of Life/VB/Conways Game of Life/program/gui/exDGV.vb
''' <summary>
''' Extended DataGridView 
''' DoubleBuffered. Restricts user selection of cells to facilitate seamless highlighting line drawing.
''' </summary>
''' <remarks></remarks>
Public Class exDGV
    Inherits DataGridView

    Dim WM_LBUTTONDOWN As Integer = &H201
    Dim WM_LBUTTONDBLCLK As Integer = &H203
    Dim WM_KEYDOWN As Integer = &H100

    Public Sub New()
        Me.DoubleBuffered = True
    End Sub

    Protected Overrides Sub OnRowPrePaint(ByVal e As System.Windows.Forms.DataGridViewRowPrePaintEventArgs)
        e.PaintParts = e.PaintParts And Not DataGridViewPaintParts.Focus
        MyBase.OnRowPrePaint(e)
    End Sub

    Protected Overrides Sub WndProc(ByRef m As System.Windows.Forms.Message)
        If m.Msg = WM_LBUTTONDBLCLK OrElse m.Msg = WM_KEYDOWN OrElse m.Msg = WM_LBUTTONDOWN Then
            Return
        End If
        MyBase.WndProc(m)
    End Sub

End Class

File: Conway's Game of Life/[C#]-Conway's Game of Life/C#/Conways Game of Life cs/app/gui/exDGV.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace Conways_Game_of_Life_cs
{
    /// <summary>
    /// Extended DataGridView 
    /// DoubleBuffered. Restricts user selection of cells to facilitate seamless highlighting line drawing.
    /// </summary>
    /// <remarks></remarks>
    public class exDGV : DataGridView
    {

        int WM_LBUTTONDOWN = 0x201;
        int WM_LBUTTONDBLCLK = 0x203;

        int WM_KEYDOWN = 0x100;
        public exDGV()
        {
            this.DoubleBuffered = true;
        }

        protected override void OnRowPrePaint(System.Windows.Forms.DataGridViewRowPrePaintEventArgs e)
        {
            DataGridViewPaintParts paintParts = (int)e.PaintParts - DataGridViewPaintParts.Focus;
            e.PaintParts = paintParts;
            base.OnRowPrePaint(e);
        }

        protected override void WndProc(ref System.Windows.Forms.Message m)
        {
            if (m.Msg == WM_LBUTTONDBLCLK || m.Msg == WM_KEYDOWN || m.Msg == WM_LBUTTONDOWN)
            {
                return;
            }
            base.WndProc(ref m);
        }

    }
}
